cpp/ql/lib/semmle/code/cpp/models/interfaces/NonThrowing.qll

```diff
@@ -1,11 +0,0 @@
-/**
- * Provides an abstract class for modeling functions that never throw.
- */
-
-import semmle.code.cpp.Function
-import semmle.code.cpp.models.Models
-
-/**
- * A function that is guaranteed to never throw.
- */
-abstract class NonThrowingFunction extends Function { }
```

Comment:
You cannot just delete files that are not in non-implementation or internal directories. These will need to go through a deprecation period. Could you discuss with @MathiasVP what the correct approach here should be.

Reply:
Indeed, we have a 1-ish year deprecation period for non-internal QL things. So we need to:
- Mark this class as `deprecated`
- Keep the functionality. i.e., extending `NonThrowingFunction` should do something equivalent to what it did before

And then it'll be deleted by someone in a PR a year from now.

Reply:
I was thinking about that after I submitted, fixed.

Reply:
Can't we just do:
```
private import Throwing as T

deprecated class NonThrowingFunction = T::NonThrowingFunction;
```
and write in the change note that (a) the class is deprecated, and (b) requires a new member to be implemented.

Reply:
Or is that too dirty?

Reply:
We can definitely do that, yes!

Reply:
We can do this, but I had avoided changes like this because any existing use of the old NonThrowingFunction since it will now require the member predicate be defined. I thought the point of deprecating vs deleting was to not completely break existing builds. If you are actually ok with that @jketema  I'm ok with it.

Reply:
I suppose another option is to just make the deprecated version extend the new throwing function mechanic and set the exception type predicate to be any type (matching the prior intended behavior).